### PR TITLE
#1479 added bom to schema

### LINKED ISSUE #1479: [BOM] - Add Schema for Bill of Materials
### Description

We wanna create the schema for the new bill of materials feature we're adding to Finishline

### Acceptance Criteria

- The Schema described below has been added to the schema.prisma file.
- A migration has been created from the schema changes (Run yarn prisma:migrate once you're done making changes in schema.prisma)
- Any associated tests, seed file, etc. are fixed if the changes cause errors in them.

### Proposed Solution

Schema Changes:
#### Add an Enum
MaterialStatus:
Received
Ordered
Shipped
Unordered

#### Add an Enum
Unit:
MM
FT
IN

#### Add an Assembly Model with: 
- assemblyId (uuid)
- name
- pdmFileName (string)
- dateDeleted (optional)
- userDeleted (optional)
- dateCreated 
- userCreated
-   wbsElement (relation)
- an array of Materials (association)
#### Add a Material Model with:
- modelId (uuid)
- name (string)
-  wbsElement (relation)
- dateDeleted (optional)
- userDeleted (optional)
- dateCreated 
- userCreated
- an Assembly (optional, relation)
- status MaterialStatus (enum)
- materialType MaterialType (relation)
- manufacturer Manufacturer (relation)
- manufacturerPartNumber (string)
- pdmFileName (string, optional)
- Quantity (int)
- quantityUnit (Unit - optional)
- Price (int) - this will be in cents
- Subtotal (int) - this will be in cents
- linkUrl (string)
- notes (string)
#### Add a MaterialType Model with:
- name (string) - should be unique/ use it as the id
- dateCreated
- creatorId (int)
- array of materials Material (relation)
#### Add a Manufacturer Model with:
- name (string) - should be unique/ use it as the id
- dateCreated
- creatorId (int)
- array of materials Material (relation)

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by walker-sean**
## Changes

Added BOM enums and models to the prisma schema

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1479

### CODE REVIEW

File: src/backend/src/prisma/schema.prisma [hunk 1]
```diff
@@ -451,3 +457,72 @@ model User_Secure_Settings {
...
 }
+
+enum Material_Status {
+  RECEIVED
+  ORDERED
+  SHIPPED
+  UNORDERED
+}
+
+enum Unit {
```
Comment (Peyton-McKee) [inline src/backend/src/prisma/schema.prisma:468]: I think unit should be a model that they can add to

**Review by RChandler234 — CHANGES_REQUESTED** [review]
you should delete the two migrations and make one new one that combines the changes from both of them

File: src/backend/src/prisma/schema.prisma [hunk 2]
```diff
@@ -451,3 +457,72 @@ model User_Secure_Settings {
...
+  dateCreated   DateTime
+  userCreated   User        @relation(fields: [userCreatedId], references: [userId], name: "assemblyCreator")
+  userCreatedId Int
+  wbsElement    WBS_Element @relation(fields: [wbsElementId], references: [wbsElementId])
+  wbsElementId  Int
+  materials     Material[]
+}
+
+model Material {
+  modelId          String          @id @default(uuid())
```
Comment (RChandler234) [inline src/backend/src/prisma/schema.prisma:489]: this should be materialId, that's my b for putting it wrong in the ticket

File: src/backend/src/prisma/schema.prisma [hunk 3]
```diff
@@ -451,3 +457,72 @@ model User_Secure_Settings {
...
+  materialType     Material_Type   @relation(fields: [materialTypeName], references: [name])
+  manufacturer     Manufacturer    @relation(fields: [manufacturerName], references: [name])
+  pdmFileName      String?
+  quantity         Int
+  quantityUnit     Unit?           @relation(fields: [unitName], references: [name])
+  price            Int
+  subtotal         Int
+  linkUrl          String
+  notes            String
+  materialTypeName String
```
Comment (RChandler234) [inline src/backend/src/prisma/schema.prisma:511]: can you put these by the relations that they're for?

File: src/backend/src/prisma/schema.prisma [hunk 4]
```diff
@@ -451,3 +457,72 @@ model User_Secure_Settings {
...
+  userDeletedId Int?
+  dateCreated   DateTime
+  userCreated   User        @relation(fields: [userCreatedId], references: [userId], name: "assemblyCreator")
+  userCreatedId Int
+  wbsElement    WBS_Element @relation(fields: [wbsElementId], references: [wbsElementId])
+  wbsElementId  Int
+  materials     Material[]
+}
+
+model Material {
```
Comment (RChandler234) [inline src/backend/src/prisma/schema.prisma:488]: you're missing manufacturerPartNumber on the Material model

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**